### PR TITLE
[feature] Fix TF tests, add --torch CLI option, allow run TF without torch installed

### PR DESCRIPTION
**Pull request by ervteng**
### Proposed change(s)

This PR does three main things:

- Adds a `--torch` CLI option that enables PyTorch, and a `framework:` option in the trainer configuration YAML. 
- Allows the code to run in TF mode without `torch` installed, and throws the appropriate exception when you try to run it.
- Fixes the TF tests. 

### Types of change(s)

- [x] Bug fix
- [x] New feature
- [ ] Code refactor
- [ ] Breaking change
- [ ] Documentation update
- [ ] Other (please describe)

### Checklist
- [ ] Added tests that prove my fix is effective or that my feature works
- [ ] Updated the [changelog](https://github.com/Unity-Technologies/ml-agents/blob/master/com.unity.ml-agents/CHANGELOG.md) (if applicable)
- [ ] Updated the [documentation](https://github.com/Unity-Technologies/ml-agents/tree/master/docs) (if applicable)
- [ ] Updated the [migration guide](https://github.com/Unity-Technologies/ml-agents/blob/master/docs/Migrating.md) (if applicable)

### CODE REVIEW

File: ml-agents/mlagents/trainers/cli_utils.py [hunk 1]
```diff
@@ -168,6 +168,13 @@ def _create_parser() -> argparse.ArgumentParser:
         action=DetectDefaultStoreTrue,
         help="Forces training using CPU only",
     )
+    argparser.add_argument(
+        "--torch",
+        default=False,
+        action=DetectDefaultStoreTrue,
+        help="(Experimental) Use the PyTorch framework instead of TensorFlow. Install PyTorch "
```
Comment (vincentpierre) [inline ml-agents/mlagents/trainers/cli_utils.py:175]: Is there a way to hide the help, or do we not care if the users see this?

File: ml-agents/mlagents/trainers/settings.py [hunk 2]
```diff
@@ -546,6 +551,7 @@ def _set_default_hyperparameters(self):
     threaded: bool = True
     self_play: Optional[SelfPlaySettings] = None
     behavioral_cloning: Optional[BehavioralCloningSettings] = None
+    framework: FrameworkType = FrameworkType.TENSORFLOW
```
Comment (vincentpierre) [inline ml-agents/mlagents/trainers/settings.py:554]: There are 2 ways to have a pytorch trainer then, with `--torch` or by adding `framework: torch` to the config file.

Comment (ervteng) [reply ml-agents/mlagents/trainers/settings.py:554]: Yep, that's true with every CLI option - it goes into the YAML somehow. That way a run can be specified entirely from a YAML file - this is for cloud training and for reproducibility.

File: ml-agents/mlagents/trainers/cli_utils.py (hunk 1 above)
Comment (ervteng) [reply ml-agents/mlagents/trainers/cli_utils.py:175]: Yes - `argparse.SUPPRESS`. I can hide it. Should we?

Comment (vincentpierre) [reply ml-agents/mlagents/trainers/cli_utils.py:175]: Let's not hide it then

**Pull request merged**